### PR TITLE
Adds OTA Support to CY8CKIT_064S0S2_4343W

### PR DESCRIPTION
**Pull request by RaymondNgun**
Adds OTA Support to CY8CKIT_064S0S2_4343W

Description
-----------
Patch set introduces OTA support to the kit. However, the "phone home" feature is absent.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/cypress/psoc6/psoc6make/make/scripts/sign_script.bash [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+# This file is used by make to create the build commands to sign an OTA image
+#
+# Modify at your peril !
+#
+# The output files will be in the same directory as the .elf, .hex, etc files
+#
+(set -o igncr) 2>/dev/null && set -o igncr; # this comment is required
+set -e
+
+# Arguments
```
Comment (gkwicker) [inline vendors/cypress/psoc6/psoc6make/make/scripts/sign_script.bash:12]: Ideally you should be using command-line options, verifying arguments, and providing help if they aren't correct.  Here's a [good stack overflow post](https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash) on how to do this in `bash`.

File: vendors/cypress/psoc6/psoc6make/make/scripts/sign_tar.bash [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+#
+# The output files will be in the same directory as the .elf, .hex, etc files
+#
+(set -o igncr) 2>/dev/null && set -o igncr; # this comment is required
+set -e
+
+# Arguments
+# We have a lot
+#
+CY_OUTPUT_PATH=$1
```
Comment (gkwicker) [inline vendors/cypress/psoc6/psoc6make/make/scripts/sign_tar.bash:15]: See previous comment

**Comment by RaymondNgun** [comment]
> Why does this PR have changes in `vendors/espressif/esp-idf`?

That's a good question. Removed. Apologies.

File: vendors/cypress/psoc6/psoc6make/make/scripts/sign_script.bash (hunk 1 above)
Comment (RaymondNgun) [reply vendors/cypress/psoc6/psoc6make/make/scripts/sign_script.bash:12]: Thanks for the feedback.  I will submit an internal ticket to enhance this. It won't be done and verified within the time frame needed here however.

**Pull request merged**